src/irc.nim

```diff
@@ -333,6 +347,14 @@ proc newIrc*(address: string, port: Port = 6667.Port,
   result.userList = initTable[string, UserList]()
   result.eventsQueue = initDeque[IrcEvent]()
 
+  when defined(ssl):
+    if ssl:
+      try:
+        sslContext.wrapSocket(result.sock)
+      except:
+        result.sock.close()
+        raise getCurrentException()
```

Comment:
``raise`` without anything does re-raise the current exception and should be preferred.